src/phoenix/trace/dsl/query.py

```diff
@@ -105,61 +87,155 @@ def from_dict(cls, obj: Mapping[str, Any]) -> "Projection":
+        return name + self._tmp_suffix
+
+
+@dataclass(frozen=True)
+class Explosion(_HasTmpSuffix, Projection):
     kwargs: Mapping[str, str] = field(default_factory=lambda: MappingProxyType({}))
     primary_index_key: str = "context.span_id"
 
-    position_prefix: str = field(init=False, repr=False)
-    primary_index: Projection = field(init=False, repr=False)
+    _position_prefix: str = field(init=False, repr=False)
+    _primary_index: Projection = field(init=False, repr=False)
+    _array_tmp_col_label: str = field(init=False, repr=False)
+    """For sqlite we need to store the array in a temporary column to be able
+    to explode it later in pandas. `_array_tmp_col_label` is the name of this
+    temporary column. The temporary column will have a unique name
+    per instance.
+    """
 
     def __post_init__(self) -> None:
         super().__post_init__()
         position_prefix = _PRESCRIBED_POSITION_PREFIXES.get(self.key, "")
-        object.__setattr__(self, "position_prefix", position_prefix)
-        object.__setattr__(self, "primary_index", Projection(self.primary_index_key))
+        object.__setattr__(self, "_position_prefix", position_prefix)
+        object.__setattr__(self, "_primary_index", Projection(self.primary_index_key))
+        object.__setattr__(self, "_array_tmp_col_label", f"__array_tmp_col_{random()}")
 
     @cached_property
-    def index_keys(self) -> Tuple[str, str]:
-        return (self.primary_index.key, f"{self.position_prefix}position")
-
-    def with_primary_index_key(self, primary_index_key: str) -> "Explosion":
-        return replace(self, primary_index_key=primary_index_key)
-
-    def __call__(self, span: Span) -> Iterator[Dict[str, Any]]:
-        if not isinstance(seq := self.value(span), Iterable):
-            return
-        has_mapping = False
-        for item in seq:
-            if isinstance(item, Mapping):
-                has_mapping = True
-                break
-        if not has_mapping:
-            for i, item in enumerate(seq):
-                if item is not None:
-                    yield {
-                        self.key: item,
-                        self.primary_index.key: self.primary_index(span),
-                        f"{self.position_prefix}position": i,
-                    }
-            return
-        for i, item in enumerate(seq):
-            if not isinstance(item, Mapping):
-                continue
-            record = (
-                {name: item.get(key) for name, key in self.kwargs.items()}
-                if self.kwargs
-                else dict(item)
+    def index_keys(self) -> List[str]:
+        return [self._primary_index.key, f"{self._position_prefix}position"]
+
+    def with_primary_index_key(self, _: str) -> "Explosion":
+        print("`.with_primary_index_key(...)` is deprecated and wil be removed in the future.")
+        return self
+
+    def update_sql(self, sql: Select[Any], dialect: str) -> Select[Any]:
```

Comment:
Suggestion: You could use `Literal` here.

Reply:
good idea